src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,59 +1,67 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{T<:AbstractProjTTN}
+  terms::Vector{T}
+
+  function ProjTTNSum(terms::Vector{<:AbstractProjTTN})
+    return new{eltype(terms)}(terms)
   end
 end
 
-copy(P::ProjTTNSum) = ProjTTNSum(copy.(P.pm))
+terms(P::ProjTTNSum) = P.terms
 
-ProjTTNSum(ttnos::Vector{<:TTN}) = ProjTTNSum([ProjTTN(M) for M in ttnos])
+copy(P::ProjTTNSum) = ProjTTNSum(copy.(terms(P)))
 
-ProjTTNSum(Ms::TTN{V}...) where {V} = ProjTTNSum([Ms...])
+ProjTTNSum(operators::Vector{<:AbstractTTN}) = ProjTTNSum(ProjTTN.(operators))
 
-on_edge(P::ProjTTNSum) = on_edge(P.pm[1])
+on_edge(P::ProjTTNSum) = on_edge(terms(P)[1])
 
-nsite(P::ProjTTNSum) = nsite(P.pm[1])
+nsite(P::ProjTTNSum) = nsite(terms(P)[1])
 
 function set_nsite(Ps::ProjTTNSum, nsite)
-  return ProjTTNSum(map(M -> set_nsite(M, nsite), Ps.pm))
+  return ProjTTNSum(map(p -> set_nsite(p, nsite), terms(Ps)))
 end
 
-underlying_graph(P::ProjTTNSum) = underlying_graph(P.pm[1])
+underlying_graph(P::ProjTTNSum) = underlying_graph(terms(P)[1])
 
-Base.length(P::ProjTTNSum) = length(P.pm[1])
+Base.length(P::ProjTTNSum) = length(terms(P)[1])
 
-sites(P::ProjTTNSum) = sites(P.pm[1])
+sites(P::ProjTTNSum) = sites(terms(P)[1])
 
-incident_edges(P::ProjTTNSum) = incident_edges(P.pm[1])
+incident_edges(P::ProjTTNSum) = incident_edges(terms(P)[1])
 
-internal_edges(P::ProjTTNSum) = internal_edges(P.pm[1])
+internal_edges(P::ProjTTNSum) = internal_edges(terms(P)[1])
 
 function product(P::ProjTTNSum, v::ITensor)::ITensor
-  Pv = product(P.pm[1], v)
-  for n in 2:length(P.pm)
-    Pv += product(P.pm[n], v)
+  Pv = product(terms(P)[1], v)
+  for n in 2:length(terms(P))
+    Pv += product(terms(P)[n], v)
+  end
+  return Pv
+end
+
+function contract(P::ProjTTNSum, v::ITensor)::ITensor
+  Pv = contract(terms(P)[1], v)
+  for n in 2:length(terms(P))
+    Pv += contract(terms(P)[n], v)
   end
   return Pv
 end
 
 function Base.eltype(P::ProjTTNSum)
-  elT = eltype(P.pm[1])
-  for n in 2:length(P.pm)
-    elT = promote_type(elT, eltype(P.pm[n]))
+  elT = eltype(terms(P)[1])
+  for n in 2:length(terms(P))
+    elT = promote_type(elT, eltype(terms(P)[n]))
   end
   return elT
```

Comment:
I think this should work:
```suggestion
  return mapreduce(eltype, promote_type, terms(P))
```